internal/k8s/controllers/config_controller_test.go

```diff
@@ -0,0 +1,265 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1beta1 "go.universe.tf/metallb/api/v1beta1"
+	v1beta2 "go.universe.tf/metallb/api/v1beta2"
+	"go.universe.tf/metallb/internal/config"
+	metallbcfg "go.universe.tf/metallb/internal/config"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestConfigController(t *testing.T) {
+	tests := []struct {
+		desc            string
+		handlerRes      SyncState
+		failReconcile   bool
+		callForceReload bool
+		validResources  bool
+	}{
+		{
+			desc:            "handler returns SyncStateSuccess, valid resources",
+			handlerRes:      SyncStateSuccess,
+			failReconcile:   false,
+			callForceReload: false,
+			validResources:  true,
```

Comment:
nit: move validResources above, as it's an "input" for the tests, so looking at each test, it's visible what's the set of inputs and what's the expectation

Reply:
moved.